staging/src/kubevirt.io/client-go/api/v1/types.go

```diff
@@ -1380,6 +1381,29 @@ type DeveloperConfiguration struct {
 	UseEmulation           bool              `json:"useEmulation,string,omitempty"`
 }
 
+// PermittedHostDevices holds inforamtion about devices allowed for passthrough
+// +k8s:openapi-gen=true
+type PermittedHostDevices struct {
+	PciHostDevices  []PciHostDevice      `json:"pciDevices,omitempty"`
+	MediatedDevices []MediatedHostDevice `json:"mdevs,omitempty"`
+}
+
+// PciHostDevice represents a host PCI device allowed for passthrough
+// +k8s:openapi-gen=true
+type PciHostDevice struct {
+	Selector                 string `json:"pciVendorSelector"`
+	ResourceName             string `json:"resourceName"`
+	ExternalResourceProvider bool   `json:"externalResourceProvider,omitempty"`
+}
+
+// MediatedHostDevice represents a host mediated device allowed for passthrough
+// +k8s:openapi-gen=true
+type MediatedHostDevice struct {
+	Selector                 string `json:"mdevNameSelector"`
+	ResourceName             string `json:"resourceName"`
+	ExternalResourceProvider bool   `json:"externalResourceProvider,omitempty"`
```

Comment:
What's thought process behind this `externalResourceProvider` type?

Is it primarily to limit the user who is creating VMs to only be capable of requesting host resources explicitly defined by the admin in an allow list? I see it useful as a gating mechanism to prevent users from attempting to attach devices that aren't necessarily compatible with the defined KubeVirt interface. I just wasn't sure if there was something more to this as well.

Reply:
`ExternalResourceProvider` is an indication that the device plugin for this resource is being provided externally, not the generic one we've created for virt-handler. (for example the NVIDIA device-plugin) - however, as you said, we will still control which devices are allowed to be used in the cluster.